### PR TITLE
Adds Github build verification

### PR DESCRIPTION
**Pull request by vogella**
The build currently fails and complains about missing version increases, see https://github.com/vogellacompany/m2e-core/runs/4455549109?check_suite_focus=true#step:6:6755

@mickaelistria @laeubi what to do here? Send separate pull requests to increase the version? Or is the check incorrect, the Jenkins build seems not to care about this.

### CODE REVIEW

**Comment by vogella** [comment]
Error:  Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:2.5.0:compare-version-with-baselines (check-no-version-regression) on project org.eclipse.m2e.maven.runtime.slf4j.simple: Only qualifier changed for (org.eclipse.m2e.maven.runtime.slf4j.simple/1.18.1.20211208-1002). Expected to have bigger x.y.z than what is available in baseline (1.18.1.20211008-0657) -> [Help 1]
Error:  Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:2.5.0:compare-version-with-baselines (check-no-version-regression) on project org.eclipse.m2e.model.edit: Only qualifier changed for (org.eclipse.m2e.model.edit/1.18.2.20211208-1002). Expected to have bigger x.y.z than what is available in baseline (1.18.2.20211002-1029) -> [Help 1]
Error:  Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:2.5.0:compare-version-with-baselines (check-no-version-regression) on project org.eclipse.m2e.logback.configuration: Only qualifier changed for (org.eclipse.m2e.logback.configuration/1.16.3.20211208-1002). Expected to have bigger x.y.z than what is available in baseline (1.16.3.20211002-1029) -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - name: Generated Meta data
+      run: mvn generate-sources -f m2e-maven-runtime -Pgenerate-osgi-metadata -Dtycho.mode=maven
+    - name: Build m2e-core
+      run: mvn clean verify --file pom.xml -fae
```
Comment (laeubi) [inline .github/workflows/maven.yml:35]: You should use -Dmaven.test.failure.ignore=true instead and publish the test results as done in 
https://github.com/eclipse/tycho/blob/master/.github/workflows/maven.yml#L46
then another action can publish the test results see https://github.com/eclipse/tycho/blob/master/.github/workflows/check.yml

Or simply not use fae if you want to fail fast (what generally is better if we are not talking about integration tests)

**Review by laeubi — CHANGES_REQUESTED** [review]

**Review by mickaelistria — CHANGES_REQUESTED** [review]
We don't disable failing tests, we either fix them or live with the annoyance they cause.

**Comment by mickaelistria** [comment]
>  Or is the check incorrect, the Jenkins build seems not to care about this.

The check is correct and Jenkins build cares about it. It's strange that bundles that were not modified are now treated as modified and generating new qualifiers. Isn't there an extra commit modifying them on your end?

**Comment by vogella** [comment]
Thanks @laeubi and @mickaelistria I adjusted the commit and removed the removal of the failing test. AFAICS no other commits modified the bundles with the new qualifiers.

**Comment by vogella** [comment]
Test failure is unrelated to this change.

**Comment by mickaelistria** [comment]
thanks!

**Comment by HannesWell** [comment]
Sorry for the question, but what is the advantage of having a Jenkins and GH-Actions build in parallel that do the same?

**Comment by vogella** [comment]
Jenkins was not reliable in the past and AFAIK GitHub actions are faster

Hannes Wellmann ***@***.***> schrieb am Mi., 8. Dez. 2021,
20:59:

**Comment by HannesWell** [comment]
> Jenkins was not reliable in the past and AFAIK GitHub actions are faster

Understand. Thanks for the explanation.

**Comment by mickaelistria** [comment]
Currently, it's the opposite; GitHub Actions build is less reliable than Jenkins; it shows an unexpected failure.

**Comment by vogella** [comment]
Which one? I checked multiple pull request and both Jenkins and build action fail. And #341 shows also in my local build.

**Comment by laeubi** [comment]
See https://github.com/eclipse-m2e/m2e-core/runs/4457287427 for example fails with

> Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:2.5.0:compare-version-with-baselines (check-no-version-regression) on project org.eclipse.m2e.maven.runtime.slf4j.simple: Only qualifier changed for (org.eclipse.m2e.maven.runtime.slf4j.simple/1.18.1.20211208-1248). Expected to have bigger x.y.z than what is available in baseline (1.18.1.20211008-0657) -> [Help 1]

That is not the test-failure like on the Ci server.

**Comment by laeubi** [comment]
The ci-build seems  to use xvnc to execute the core build, could this be an issue?

**Comment by mickaelistria** [comment]
I don't think it's related to Xvnc. The qualifier is derived from Git history (last modification timestamp). I don't know what can cause it to be different on GH Actions.

**Comment by HannesWell** [comment]
In my latest PR GH-Actions failed again (reporting the version regressions I cannot explain), while Jenkins succeeds.
Therefore I suggest to revert this PR.

**Comment by mickaelistria** [comment]
> Therefore I suggest to revert this PR.

No objection against a revert. The Jenkins CI does the job, so if the Github Actions cause annoyance and no-one plans to get them working, then it's better to not have it.

**Pull request merged**